From 4ec074dcac43e7c149c1301e27f6571612803802 Mon Sep 17 00:00:00 2001
From: Yi Zhao <yi.zhao@windriver.com>
Date: Fri, 3 Jul 2020 09:07:29 +0800
Subject: [PATCH] policy/modules/services/inetd: make ipsec domain MLS trusted
 for reading from files up to its clearance

* Allow ipsec_supervisor_t to search /run/systemd/journal
* Allow ipsec_t to search /run/systemd/journal
* Allow ipsec_t to map /etc/ipsec.secrets

Fixes:
avc:  denied  { search } for  pid=277 comm="starter" name="journal"
dev="tmpfs" ino=10990
scontext=system_u:system_r:ipsec_supervisor_t:s0-s15:c0.c1023
tcontext=system_u:object_r:syslogd_runtime_t:s15:c0.c1023 tclass=dir
permissive=0

avc:  denied  { search } for  pid=288 comm="charon" name="journal"
dev="tmpfs" ino=10990 scontext=system_u:system_r:ipsec_t:s0-s15:c0.c1023
tcontext=system_u:object_r:syslogd_runtime_t:s15:c0.c1023 tclass=dir
permissive=0

avc:  denied  { map } for  pid=288 comm="charon"
path="/etc/ipsec.secrets" dev="vda" ino=400
scontext=system_u:system_r:ipsec_t:s0-s15:c0.c1023
tcontext=system_u:object_r:ipsec_key_file_t:s0 tclass=file permissive=0

Upstream-Status: Pending

Signed-off-by: Yi Zhao <yi.zhao@windriver.com>
---
 policy/modules/system/ipsec.te | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/policy/modules/system/ipsec.te b/policy/modules/system/ipsec.te
index bdb077ac8..646a1e2a4 100644
--- a/policy/modules/system/ipsec.te
+++ b/policy/modules/system/ipsec.te
@@ -97,6 +97,7 @@ read_files_pattern(ipsec_t, ipsec_conf_file_t, ipsec_conf_file_t)
 read_lnk_files_pattern(ipsec_t, ipsec_conf_file_t, ipsec_conf_file_t)
 
 allow ipsec_t ipsec_key_file_t:dir list_dir_perms;
+allow ipsec_t ipsec_key_file_t:file map;
 manage_files_pattern(ipsec_t, ipsec_key_file_t, ipsec_key_file_t)
 read_lnk_files_pattern(ipsec_t, ipsec_key_file_t, ipsec_key_file_t)
 
@@ -185,6 +186,8 @@ sysnet_domtrans_ifconfig(ipsec_t)
 userdom_dontaudit_use_unpriv_user_fds(ipsec_t)
 userdom_dontaudit_search_user_home_dirs(ipsec_t)
 
+mls_file_read_to_clearance(ipsec_t)
+
 optional_policy(`
 	seutil_sigchld_newrole(ipsec_t)
 ')
@@ -498,6 +501,8 @@ logging_send_syslog_msg(ipsec_supervisor_t)
 
 miscfiles_read_localization(ipsec_supervisor_t)
 
+mls_file_read_to_clearance(ipsec_supervisor_t)
+
 optional_policy(`
 	modutils_domtrans(ipsec_supervisor_t)
 ')
-- 
2.17.1

